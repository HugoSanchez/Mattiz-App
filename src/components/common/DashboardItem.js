import React from 'react';
import { View, Text, Image, Dimensions } from 'react-native';

// General Styles & Colors
import GS from '../../styles'
import colors from '../../constants/colors'

// Number parser 
const numeral = require('numeral');

// Dims. 
const height = Dimensions.get('window').height

// Props:
// @containerHeight: object, optional; 
// @iconBox: object, optional.
// @imageStyle: object, optional.
// @image: component, optional.
// @uri: sting, required. 
// @title: string, required.
// @amount: string, required.
// @percentage: string, required.

const DashboardItem = props => {
    return (
        <View style={[ styles.container, props.containerHeight ]}>
            <View style={ GS.container }>
                <View style={[ styles.iconBox, props.iconBox ]}>
                    {
                        props.image ?
                        props.image
                        :
                        <Image 
                            style={[ styles.imageStyle, props.imageStyle ]}
                            source={{ uri: props.uri }}
                        />
                    }
                </View>
            </View>
                            
            <View style={ styles.textBox }>
                <Text style={ styles.textStyle }> 
                    {
                        props.title
                    }
                </Text>
            </View>
                            
            <View style={ styles.numbersBox }>
                <View style={ GS.container }>
                    <Text style={ styles.amountStyle }>
                        {  numeral(props.amount).format('0,0.00') } â‚¬
                    </Text> 
                </View>
                 <View style={ GS.container }>
                    <Text style={[ 
                        styles.percentageStyles, 
                        props.positive ? 
                        { color: colors.numbersGreen } 
                        : 
                        { color: colors.numbersRed }
                    ]}>
                        {
                            props.percentage
                        }
                    </Text> 
                </View>
            </View>

        </View>
    );
}

const styles = {
    container: {
        height: height * 0.08, 
        flexDirection: 'row', 
        paddingLeft: '2%'
    },
    iconBox: { 
        height: height * 0.05, 
        width: height * 0.05, 
        borderRadius: (height * 0.05) / 2, 
        alignItems: 'center', 
        justifyContent: 'center', 
        backgroundColor: '#FFF'
    },
    imageStyle: { 
        height: height * 0.04, 
        width: height * 0.04, 
        borderRadius: (height * 0.04) / 2 
    },
    textBox: { 
        flex: 3, 
        justifyContent: 'center' 
    },
    textStyle: { 
        fontFamily: 'Raleway-Regular', 
        fontSize: 18 
    },
    numbersBox: { 
        flex: 2 
    },
    amountStyle: [ 
        GS.extraSmallLightNumber, 
        { marginTop: '15%'} 
    ],
    percentageStyles: [ 
        GS.extraSmallBoldNumber, 
        { 
            color: colors.numbersGreen, 
            fontSize: 12, 
            marginBottom: '10%' 
        } 
    ]
};

export default DashboardItem;