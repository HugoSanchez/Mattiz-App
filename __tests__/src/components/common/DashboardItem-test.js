import 'react-native'
import React from 'react'
import DashboardItem from '../../../../src/components/common/DashboardItem'
import {render, fireEvent} from '@testing-library/react-native'

test('Renders with title', () => {
	const {container} = render(<DashboardItem />)
	expect(container.children[0]).toMatchSnapshot()
})
