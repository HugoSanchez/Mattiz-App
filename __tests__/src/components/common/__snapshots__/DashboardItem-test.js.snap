// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "height": 106.72,
        "paddingLeft": "2%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderRadius": 33.35,
            "height": 66.7,
            "justifyContent": "center",
            "width": 66.7,
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 26.68,
              "height": 53.36,
              "width": 53.36,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#154053",
          "fontFamily": "Rajdhani-Medium",
          "fontSize": 18,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#154053",
              "fontFamily": "Rajdhani-Regular",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "flex-end",
              "color": "#000",
              "marginRight": "15%",
              "marginTop": "15%",
            },
          ]
        }
      >
        0.00
         â‚¬
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Array [
              Object {
                "color": "#154053",
                "fontFamily": "Rajdhani-Bold",
                "fontSize": 16,
              },
              Object {
                "alignSelf": "flex-end",
                "color": "#C4E19C",
                "fontSize": 14,
                "marginBottom": "10%",
                "marginRight": "20%",
              },
            ],
            Object {
              "color": "#C88383",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
